protobufs/livekit_models.proto

```diff
@@ -286,6 +287,14 @@ message TranscriptionSegment {
   string language = 6;
 }
 
+message ChatMessage {
+  string id = 1; // populated by server
+  int64 timestamp = 2; // populated by  server
+  int64 edit_timestamp = 3; // populated by server
+  string message = 4;
+  bool deleted = 5; // true to remove message
```

Comment:
Shouldn't the sending client be populating these fields? What would the sending client message look like? Just the `message`?

And server would populate the rest before forwarding? But, that will cause issues for edits. How would the server know if a message has been edited if it is the one generating the unique message id.

Reply:
ah, not sure why the `id` is also generated by the server. 
Intention was only for the `timestamps` to be populated by the server. 
`id, message and deleted` would be set by the client. 
Does that make more sense?

Reply:
Got it.

I am guessing `deleted` also should be populated by sending client?

A few more questions
1. How many messages (ids) should server hold on to? Feels like sending client should populate the time stamps and server will just normalise it to server clock maybe?
2. If server is holding on to messages, should it replay them on a new participant join?

Reply:
> I am guessing deleted also should be populated by sending client?

Ignore that, thought that also said it was populated by server, but no, the comment does not say that.